api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/policy/PolicyLoader.java

```diff
@@ -0,0 +1,662 @@
+                    loadRulesForDataSet(uri, rules, dataSetName);
+                }
+            }
+        } catch (Exception e) {
+            return null;
+        }
+        if (rules.isEmpty()) {
+            return null;
+        }
+        DynamicPolicy policy = new DynamicPolicy(uri, priority);
+        policy.addRules(rules);
+        return policy;
+    }
+
+    public Set<String> getPolicyDataSetValues(OperationGroup og, AccessObjectType aot, String role) {
+        Set<String> values = new HashSet<>();
+        long expectedSize = 3;
+        final String queryText = getPolicyTestValuesByKeyQuery(new String[] { role },
+                new String[] { og.toString(), aot.toString() });
+        debug("SPARQL Query to get policy data set values:\n %s", queryText);
+        try {
+            rdfService.sparqlSelectQuery(queryText, new ResultSetConsumer() {
+                @Override
+                protected void processQuerySolution(QuerySolution qs) {
+                    if (!qs.contains(POLICY) || !qs.contains("keySize") || !qs.get("keySize").isLiteral()) {
+                        return;
+                    }
+                    long keySize = qs.getLiteral("keySize").getLong();
+                    if (expectedSize != keySize) {
+                        return;
+                    }
+                    if (qs.contains("valueId") && qs.get("valueId").isLiteral()) {
+                        values.add(qs.getLiteral("valueId").getString());
+                    } else if (qs.contains("value")) {
+                        RDFNode node = qs.get("value");
+                        if (node.isLiteral()) {
+                            values.add(node.asLiteral().toString());
+                        } else if (node.isResource()) {
+                            values.add(node.asResource().getURI());
+                        }
+                    }
+                }
+            });
+        } catch (RDFServiceException e) {
+            log.error(e, e);
+        }
+        return values;
+    }
+
+    public String getPolicyUriByKey(OperationGroup og, AccessObjectType aot, String role) {
+        long expectedSize = 3;
+        final String queryText = getPolicyTestValuesByKeyQuery(new String[] { role },
+                new String[] { og.toString(), aot.toString() });
+        debug("SPARQL Query to get policy data set values:\n %s", queryText);
+        String[] uri = new String[1];
+        try {
+            rdfService.sparqlSelectQuery(queryText, new ResultSetConsumer() {
+                @Override
+                protected void processQuerySolution(QuerySolution qs) {
+                    if (!qs.contains(POLICY) || !qs.get(POLICY).isResource() || !qs.contains("keySize")
+                            || !qs.get("keySize").isLiteral()) {
+                        return;
+                    }
+                    long keySize = qs.getLiteral("keySize").getLong();
+                    if (expectedSize != keySize) {
```

Comment:
why it is important that keySize is 3?

Reply:
Policies with key size of 3 (Object type, user role and operation group) are created to provide same functionality as permissions in advanced role management.
That is a safety check was done to filter only applicable policies.

Reply:
thanks for clarification